### ISSUE TITLE
Monkey patching cv2 methods crashes own scripts

### ISSUE THREAD

**Issue by janzumsteg** [ultralytics/yolov5#7285]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

PyTorch Hub

### Bug

The current master branch contains monkey patches of `cv2.imread`, `cv2.imwrite` and `cv2.imshow` (see https://github.com/ultralytics/yolov5/blob/ea72b84f5e690cb516642ce2d9ae200145b0af34/utils/general.py#L944). When importing yolov5 (e.g. via `torch.hub.load`), the respective methods in my own scripts will also be overwritten without any hints. Moreover, the signature of `cv2.imread` changed with these monkey patches resulting in `TypeError: imread() takes 1 positional argument but 2 were given`.

Related PRs:

- https://github.com/ultralytics/yolov5/pull/7139
- https://github.com/ultralytics/yolov5/pull/6979

### Minimal Reproducible Example

```
import cv2
import torch

cv2.imread("path/to/image.png", cv2.IMREAD_COLOR)  # works
torch.hub.load("ultralytics/yolov5", "yolov5m", force_reload=True)
cv2.imread("path/to/image.png", cv2.IMREAD_COLOR)  # fails
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@janzumsteg good news 😃! Your original issue may now be fixed ✅ in PR #7287. This PR updates the patched function to accept a second argument just like the original function, and the code to reproduce the error now runs without issue:

<img width="870" alt="Screenshot 2022-04-05 at 11 48 40" src="https://user-images.githubusercontent.com/26833433/161727831-9e3857e8-a34f-434d-9e02-169f7269b797.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by grzegorzk** [comment]
Hi @glenn-jocher, I have been playing with `yolov5==7.0.11` and found that monkey-patched `imread` has different signature than the one provided by opencv. Your argument is called `path` while in opencv it's called `filename`. This is resulting in my other scripts that are executed after loading yolov5 to fail, even though they are working fine if I don't use yolov5.

```python3

>>> import cv2
>>> help(cv2.imread)
Help on built-in function imread:

imread(...)
    imread(filename[, flags]) -> retval
(...)

>>> import yolov5
>>> help(cv2.imread)
Help on function imread in module yolov5.utils.general:

imread(path, flags=1)


```

I proposed fix [here](https://github.com/ultralytics/yolov5/pull/11209)

I would personally prefer if these methods are not monkey-patched each time I import yolov5 though, I can prepare a PR if this is something you would consider.

**Comment by glenn-jocher** [comment]
Hi @grzegorzk 👋, it's great to hear your feedback on the monkey-patched `imread` method. Your proposed fix in PR #11209 looks like a step in the right direction. I'll make sure to discuss this with the team to see if we can merge your changes.

Regarding the monkey-patching, I totally understand your concern. We are currently evaluating different options to reduce the impact of monkey-patching on user-defined scripts. Your willingness to submit a PR to help address this is much appreciated! We'd certainly be open to reviewing your solution. Thank you for your contribution and for helping make YOLOv5 even better 🙌!
